system/ComposerScripts.php

```diff
@@ -88,6 +88,8 @@ private static function recursiveDelete(string $directory): void
     {
         if (! is_dir($directory)) {
```

Comment:
```suggestion
        if (! file_exists($directory)) {
```

Reply:
Why suggest `file_exists()`? Will the iterator not throw an error if value passed is a file?

Reply:
On Unix, with the trailing '/', PHP search for a directory.
I tried to delete ThirdParty and `touch` a file named ThirdParty, and the function fails because of the trailing '/'.

Reply:
Oh, I missed the trailing `/`.